### DISCUSSION TITLE
Tool to validate front matter YAML in markdown files

### DISCUSSION THREAD

**Discussion by lamyergeier** [jgm/pandoc#10121] [Q&A]
Is there any tool to validate front matter YAML in markdown files?

# Note

- `yamllint` doesn't support it (refer: [[Support YAML front-matter in markdown files · Issue #161 · adrienverge/yamllint](https://github.com/adrienverge/yamllint/issues/161)](https://github.com/adrienverge/yamllint/issues/161))
- `yq` ([[mikefarah/yq: yq is a portable command-line YAML, JSON, XML, CSV, TOML and properties processor](https://github.com/mikefarah/yq)](https://github.com/mikefarah/yq)) can extract YAML front matter only if it is a valid YAML.

**Comment by jgm** [comment]
What are you trying to do?  If you want to *extract* YAML metadata, you can do that with pandoc itself.  Create a template `meta.tpl`, containing
```
$meta-json$
```
then `pandoc --template meta-json -t markdown your-document.md`.

If you want to raise an error if the YAML is malformed, pandoc's parser will do this itself.
